aten/src/ATen/native/cpu/UnaryOpsKernel.cpp

```diff
@@ -523,8 +523,8 @@ static void kaiser_window_kernel(TensorIteratorBase& iter, int64_t window_length
     using opmath_t = at::opmath_type<scalar_t>;
     const opmath_t alpha = static_cast<opmath_t>((window_length - 1) / 2.0);
     const opmath_t beta_ = static_cast<opmath_t>(beta);
-    cpu_kernel(iter, [=](scalar_t a){
-        return calc_i0(beta_ * std::sqrt(1 - std::pow((static_cast<opmath_t>(a) - alpha) / alpha, static_cast<opmath_t>(2.0)))) / calc_i0(beta_);
+    cpu_kernel(iter, [=](scalar_t a) -> scalar_t {
+        return calc_i0(beta_ * std::sqrt(std::abs(1 - std::pow((static_cast<opmath_t>(a) - alpha) / alpha, static_cast<opmath_t>(2.0))))) / calc_i0(beta_);
```

Comment:
Is this `std::abs` necessary? `a` is non-negative here and `std::pow((static_cast<opmath_t>(a) - alpha) / alpha` is always >=0, right?

Reply:
When data type is bf16, `1 - std::pow((static_cast<opmath_t>(a) - alpha) / alpha, static_cast<opmath_t>(2.0)))` may be < 0 as bf16 will lose accuracy.

Reply:
But this formula `1 - std::pow((static_cast<opmath_t>(a) - alpha) / alpha, static_cast<opmath_t>(2.0)))` is of float type when `scalar_t` is `bfloat16`, right?

Reply:
Yes. But scalar_t a has already loss accuracy. 
It indeed may generate negative values when dtype is bf16. https://github.com/pytorch/pytorch/blob/main/aten/src/ATen/native/TensorFactories.cpp#L1642

Reply:
Make sense. Thanks.